src/cryptonote_config.h

```diff
@@ -252,6 +252,7 @@ enum class hf : uint8_t {
     hf19_reward_batching,
     hf20_eth_transition,  // Temp period: registrations disabled, BLS pubkeys in proofs
     hf21_eth,             // Full transition: registrations from ETH
+    hf22_eth_beneficiary, // Add beneficiary to eth::event::Contributor (and into TX hashes which affects consensus)
```

Comment:
I don't really want to define a new hardfork for this; our hardforks generally are there to track changes on mainnet consensus, with devnet/testnet/etc. working around it -- but this feature is going to be something launching on the initial mainnet, which means mainnet won't have a meaningful hf21.

Given the plan to put out a mandatory stagenet upgrade, and only afterwards follow up with the contract update that will produce the new values, I don't see a problem (for stagenet) in introducing these retroactively into hf21.  Yes, that's a concensus breakage as soon as the first v2 one hits, but on stagenet, that's okay given the nature of the network.

Reply:
Actually yeah, I think we can omit this, even so in mainnet because you'd want to issue a MSNU that bumps the patch not a hardfork